src/semantic_hashing_demo/preprocessing.py

```diff
@@ -0,0 +1,47 @@
+""" 
+Preprocess the data and generate the hash buckets for the text reviews.
+Store the buckets for different nbits into different files.
+"""
+
+import pathlib
+
+import polars as pl
+from config import data_file, embedding_size, model, seed
+from lsh import LSH
+from utils import check_file_exists
+
+output_dir = pathlib.Path("output")
+
+
+def main():
+    # load data
+    df = pl.read_csv(data_file)
+    reviews = df.get_column("Text").to_numpy().flatten()
+    # reviews = reviews[0:10]
+
+    for nbits in [8, 16, 32, 64, 128]:
+        print(f"\n\n=====For nbits = {nbits}======\n")
+
+        # Create LSH instance
+        lsh = LSH(nbits=nbits, seed=seed, embedding_size=embedding_size)
+
+        # Generate embeddings for each review
+        embeddings = [LSH.get_embedding(review, model) for review in reviews]
```

Comment:
slight optimization: return np arrays from get_embedding, then set embeddings to the matrix, then do matrix mult for hash_vector instead of looping over every vector

Reply:
Thanks for pointing this out. Fixed [here](https://github.com/abhi3700/semantic-hashing-demo/commit/b9bb4128eac75f4e0ae2f6fb6118d9d81a3f1b99).

I actually had used batching in the [rust counterpart](https://github.com/abhi3700/semantic-hashing-demo-rs/blob/cfaf1b37fd5812ba3d934ede6ab0e00a6955a2fe/src/utils.rs#L32). But, somehow forgot to add it here.